### PR TITLE
RPC/Wallet: unloadwallet: Clarify docs/error when both the RPC request and wallet_name parameter specify a wallet

### PR DESCRIPTION
**Pull request by luke-jr**
Just documentation clarifications from #20448

### CODE REVIEW

**Comment by MarcoFalke** [comment]
cr ACK 81162ec41e5ee2ce06829dea8865925c1155cc68

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -2792,7 +2792,7 @@ static RPCHelpMan unloadwallet()
                 "Unloads the wallet referenced by the request endpoint otherwise unloads the wallet specified in the argument.\n"
                 "Specifying the wallet name on a wallet endpoint is invalid.",
                 {
-                    {"wallet_name", RPCArg::Type::STR, /* default */ "the wallet name from the RPC request", "The name of the wallet to unload."},
+                    {"wallet_name", RPCArg::Type::STR, /* default */ "the wallet name from the RPC request", "The name of the wallet to unload. Must be provided in the RPC request or this parameter (but not both)."},
```
Comment (promag) [inline src/wallet/rpcwallet.cpp:2795]: RPC request or RPC endpoint?

**Review by promag — COMMENTED** [review]
RPC description kind of says the same already:
```
Unloads the wallet referenced by the request endpoint otherwise unloads the wallet specified in the argument.
Specifying the wallet name on a wallet endpoint is invalid.
```
ACK improving current error.

File: test/functional/wallet_multiwallet.py [hunk 2]
```diff
@@ -355,7 +355,7 @@ def wallet_file(name):
         assert_raises_rpc_error(-1, "JSON value is not a string as expected", self.nodes[0].unloadwallet)
         assert_raises_rpc_error(-18, "Requested wallet does not exist or is not loaded", self.nodes[0].unloadwallet, "dummy")
         assert_raises_rpc_error(-18, "Requested wallet does not exist or is not loaded", node.get_wallet_rpc("dummy").unloadwallet)
-        assert_raises_rpc_error(-8, "Cannot unload the requested wallet", w1.unloadwallet, "w2"),
+        assert_raises_rpc_error(-8, "Both the RPC request wallet and wallet_name parameter were provided (only one allowed)", w1.unloadwallet, "w2"),
```
Comment (jonatack) [inline test/functional/wallet_multiwallet.py:358]: Please include a test with the same wallet name, e.g. `w1.unloadwallet, w1.getwalletinfo()["walletname"]`

**Comment by luke-jr** [comment]
Addressed review by @jonatack

**Comment by jonatack** [comment]
ACK e8303a08e7226cd9640001b9b9d50afd8533dc8e

**Comment by luke-jr** [comment]
...also used "endpoint" in error message and commit message...

**Comment by MarcoFalke** [comment]
review ACK b1f59d55d920d2b35269b474762f94fec87bfb16

**Comment by jonatack** [comment]
re-ACK b1f59d55d920d2b35269b474762f94fec87bfb16  per `git diff e8303a0 b1f59d5`

**Comment by MarcoFalke** [comment]
Backported in #20475

**Pull request merged**